### ISSUE TITLE
Compile errors after rev 564.

### ISSUE THREAD

**Issue by firew4lker** [OpenShot/libopenshot#42]
There is a compile error on Archlinux after revision 564. 

/usr/include/libopenshot-audio/JuceLibraryCode/modules/juce_audio_basics/../juce_core/maths/juce_MathsFunctions.h:384:12: error: ‘finite’ is not a member of ‘std’
     return std::isfinite (value);

**Comment by joebonrichie** [comment]
Also getting this error on Solus, is there something obvious i'm missing?

**Comment by joebonrichie** [comment]
Just an FYI had to rebuild with ruby bindings disabled -DENABLE_RUBY=OFF to compile. Can provide more details if this gets any attention.

**Comment by ferdnyc** [comment]
I just hit this myself, compiling with gcc 7.3.1 on Fedora 27.

The issue, surprisingly enough, is actually in libopenshot-audio, specifically one of the macros in its header files that libopenshot includes.

I do not even _remotely_ pretend to understand why this fixes the issue or even makes a difference, but the same bug was reported over at open-source-parsers/jsoncpp#214, and [a comment there](https://github.com/open-source-parsers/jsoncpp/issues/214#issuecomment-77936298) pointed me at the fix.

Here's a diff rooted in the `libopenshot-audio-0.1.5` source tree, that shows the fix:

```diff
diff -u JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h.orig JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h
--- JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h.orig	2018-02-21 03:54:09.141547407 -0500
+++ JuceLibraryCode/modules/juce_core/maths/juce_MathsFunctions.h	2018-02-21 03:54:36.394066259 -0500
@@ -381,7 +381,8 @@
    #if JUCE_WINDOWS && !JUCE_MINGW
     return _finite (value) != 0;
    #else
-    return std::isfinite (value);
+    using namespace std;
+    return isfinite (value);
    #endif
 }
 
@@ -391,7 +392,8 @@
    #if JUCE_WINDOWS && !JUCE_MINGW
     return _finite (value) != 0;
    #else
-    return std::isfinite (value);
+    using namespace std;
+    return isfinite (value);
    #endif
 }
 
```

Like I said, I don't pretend to understand why that works (I'm not a C++ guy), but if libopenshot-audio is built and installed with the headers patched in that way (or if the headers are manually modified post-install, which is how I originally tested the change), then libopenshot will compile successfully.

(If you're using the `libopenshot-0.1.5.tar.gz` release code, though, you also need to apply the fix from pull request #67 in this repo.)

**Comment by ferdnyc** [comment]
Very belated PR submitted at OpenShot/libopenshot-audio#19, when that's merged this can finally be closed. (Shutting the barn door after the horses are gone, as they say, now that libopenshot-0.2.0 is already released. But, oh well.)

**Comment by ferdnyc** [comment]
Closed with OpenShot/libopenshot-audio#37
